### ISSUE TITLE
Allow setting custom RoowKey values

### ISSUE THREAD

**Issue by manekinekko** [nestjs/azure-database#21]
This would be simple to add. Here are the steps:
1. Update https://github.com/nestjs/azure-database/blob/master/lib/table-storage/azure-table.repository.ts#L107 to:
```typescript
async create(entity: T, rowKeyValue?: string): Promise<T> {
```

2. Update https://github.com/nestjs/azure-database/blob/master/lib/table-storage/azure-table.repository.ts#L110 to:

```typescript
entity = AzureEntityMapper.createEntity<T>(entity, rowKeyValue );
```

3. update specs files.

**Comment by tabs-not-spaces** [comment]
Doing this now, but even if we include rowKeyValue into the create function, the static createEntity in https://github.com/nestjs/azure-database/blob/master/lib/table-storage/azure-table.mapper.ts#L36 needs to be updated as well, yes? otherwise it's just going to replace rowKeyValue with a GUID...

**Comment by tabs-not-spaces** [comment]
my other thought was simply modifying https://github.com/nestjs/azure-database/blob/03ac8b32fd37b31623c804384ca25e35d4feadbf/lib/table-storage/azure-table.mapper.ts#L36 to

``` typescript
static createEntity(partialDto, rowKeyValue) {
```

and updating https://github.com/nestjs/azure-database/blob/03ac8b32fd37b31623c804384ca25e35d4feadbf/lib/table-storage/azure-table.mapper.ts#L49 to:

``` TypeScript
 if (null == partialDto.RowKey) {
      entityDescriptor.RowKey._ = rowKeyValue
  }
  else {
      entityDescriptor.RowKey._  = partialDto.RowKey;
  }
```

This way, we can define a RowKey in the entity / DTO which will take precedence, or if we don't implicitly define a RowKey a GUID will be created.

**Comment by manekinekko** [comment]
closes in 070bad3de31b2dc54723bbc48470af9a084bde13
